DAO.sol

```diff
@@ -490,14 +490,22 @@ contract DAO is DAOInterface, Token, TokenSale {
             throw;
         }
 
-        if (p.newServiceProvider) {
+        if (p.newServiceProvider && now > p.votingDeadline + 28) {
             p.open = false;
             return;
         }
```

Comment:
I suggest:
```js
if (p.newCurator) {
    if (now > p.votingDeadline + 28) {
        p.open = false;
    } 
    return;
}
```

Because as it stands I can still try to execute a split Proposal and continue further on the code, which I should not be able to.